src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +395,422 @@ else if (!period.isEmpty()) {
+     *
+     * @param javadoc Javadoc root node.
+     * @return First inline tag node or null if no node is found.
+     */
+    private static DetailNode getFirstInlineTag(DetailNode javadoc) {
+        DetailNode node = null;
+        final DetailNode[] children = javadoc.getChildren();
+        for (DetailNode child: children) {
+            // If present as a children of javadoc
+            if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                node = child;
+            }
+            // If nested inside html tag
+            else if (child.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                node = getInlineTagNodeWithinHtmlElement(child);
+            }
+
+            if (node != null) {
+                break;
+            }
+        }
+        return node;
+    }
+
+    /**
+     * Returns inline javadoc tag node inside html tags from a HTML tag.
+     *
+     * @param ast html tag node.
+     * @return Inline summary javadoc tag node or null if no node is found.
+     */
+    private static DetailNode getInlineTagNodeWithinHtmlElement(DetailNode ast) {
+        DetailNode node = ast;
+        DetailNode result = null;
+        // node can never be null as it is always called when there is HTML element
+        if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+            result = node;
+        }
+        else if (node.getType() == JavadocTokenTypes.HTML_TAG) {
+            // HTML_TAG always has more than 2 children.
+            node = node.getChildren()[1];
+            result = getInlineTagNodeWithinHtmlElement(node);
+        }
+        else if (node.getType() == JavadocTokenTypes.HTML_ELEMENT
+                // Condition for SINGLETON html element which cannot contain summary node
+                && node.getChildren()[0].getChildren().length > 1) {
+            // Html elements have one tested tag before actual content inside it
+            node = node.getChildren()[0].getChildren()[1];
+            result = getInlineTagNodeWithinHtmlElement(node);
+        }
+        return result;
+    }
+
+    /**
+     * Checks if the first tag inside ast is summary tag.
+     *
+     * @param javadoc root node.
+     * @return true, if first tag is summary tag.
+     */
+    private static boolean isSummaryTag(DetailNode javadoc) {
+        final DetailNode[] child = javadoc.getChildren();
+
+        // Checking size of ast is not required, since ast contains
+        // children of Inline Tag, as at least 2 children will be present which are
+        // RCURLY and LCURLY.
+        return child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                && SUMMARY_TEXT.equals(child[1].getText());
+    }
+
+    /**
+     * Check the summary of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    private void validateSummaryTag(DetailNode ast) {
+        final String inlineSummary = getInlineSummary(ast);
+        if (inlineSummary.isEmpty()) {
```

Comment:
This will be a major change. This reduces a lot of methods but introduces multiple regexp. I had done changes but I believe we all are not a big fan of regexp. I  have not squashed the commits to make sure first that this implementation is proper so that we can revert later.

Please confirm if current implementation is correct.

Reply:
for now I expect only regexp to remove html tags to get rough understanding if there is visible content it is required only for single IF.
But ..... we probably can make method that do finding of this `containsPrintableSymbols(ast)` ... good idea, lets try it.

Reply:
Can you please explain in bit detail what this `containsPrintableSymbols(ast)` should do?

Reply:
it is same by meaning as `getVisibleContent`, problem is with term "visible". ` `(space) is visible but it not printable.
Check care that summary has printable content, printable content is visible in summary html table.
HTML tags are also not printable in HTML rendering, as they are interpretated layout/decoration.

name could be `containsHtmlPrintableSymbols`

we just need to find that summary value in javadoc tag, will be visible by user (user see only printable symbols).

Reply:
Sir, please tell is the requirement of this method for this check?

I get it that ` ` (space) is not visible but for example, let us consider a case like 
`{@summary  <p>(space).</p>}`, from `getVisibleContent` we would get ` (space).`, what my focus here is `PERIOD`. So here the existence of ` (space)` does not matter, as there would not be any violation, as there is a `period` for a summary, as you said in [#9174 comment](https://github.com/checkstyle/checkstyle/pull/9174#discussion_r612126053), "Summary that is simply `.` is good for Check."

If there is no period like in `{@summary  <p>(space)</p>}` we would get ` (space)` from `getVisibleContent` but if I just `trim()` that we get an empty string. So we are required in that case to create a new method. 

Please let me know a case where this method is required.

Reply:
https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

> 7.2 The summary fragment
Each Javadoc block begins with a brief summary fragment. This fragment is very important: it is the only part of the text that appears in certain contexts such as class and method indexes.

such indexes are html areas with summary content, so if text is not rendered, it is not visible to user.
Decision on violation should be simple. If content is visible in HTML summary table - no violation, if not visible - violation.

> `{@summary <ahref="mailto:vlad@htmlbook.ru"/>}`

will not be visible in html summary table, so violation should be on it.

> `{@summary <p> </p>}`

will aslo be rendered as no summary in HTML, so violation should be.

-----

There is another point is "what content validate by regexp as forbidden content? only visible or raw content of javadoc comment?"

I think better to validate raw content of javadoc comment to let user have more control over content of javadoc.

Reply:
>>   `{@summary <ahref="mailto:vlad@htmlbook.ru"/>}`

> will not be visible in html summary table, so violation should be on it.

Already in test case

https://github.com/checkstyle/checkstyle/blob/027e3f86b796301594300ef8b865588415f750b4/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocInlineForbidden.java#L47-L50

>> `{@summary <p> </p>}`

> will aslo be rendered as no summary in HTML, so violation should be.

Added in new test case.

> if content is visible in HTML summary table - no violation, if not visible - violation

Yup, that is only being followed.

> I think better to validate raw content of javadoc comment to let user have more control over content of javadoc.

Works, what we can do is to check the raw content that we are getting from `getInlineSummary()`, and check for the forbidden content.

Reply:
Done.